src/main/java/org/candlepin/async/tasks/ConsumerMigration.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2009 - 2024 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.async.tasks;
+
+import org.candlepin.config.ConfigProperties;
+import org.candlepin.config.Configuration;
+import org.candlepin.model.ConsumerCurator;
+import org.candlepin.model.ContentAccessCertificateCurator;
+import org.candlepin.model.IdentityCertificateCurator;
+import org.candlepin.model.Owner;
+import org.candlepin.model.OwnerCurator;
+
+import com.google.common.collect.Iterables;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+/**
+ * Class responsible for migration of consumers from anonymous orgs.
+ */
+public class ConsumerMigration {
+
+    private static final Logger log = LoggerFactory.getLogger(ConsumerMigration.class);
+
+    private final OwnerCurator ownerCurator;
+    private final ConsumerCurator consumerCurator;
+    private final IdentityCertificateCurator identityCertificateCurator;
+    private final ContentAccessCertificateCurator contentAccessCertificateCurator;
+    private final int batchSize;
+
+    @Inject
+    public ConsumerMigration(OwnerCurator ownerCurator, ConsumerCurator consumerCurator,
+        IdentityCertificateCurator identityCertificateCurator,
+        ContentAccessCertificateCurator contentAccessCertificateCurator,
+        Configuration config) {
+        this.ownerCurator = Objects.requireNonNull(ownerCurator);
+        this.consumerCurator = Objects.requireNonNull(consumerCurator);
+        this.identityCertificateCurator = Objects.requireNonNull(identityCertificateCurator);
+        this.contentAccessCertificateCurator = Objects.requireNonNull(contentAccessCertificateCurator);
+        this.batchSize = config.getInt(ConfigProperties.CONSUMER_MIGRATION_BATCH_SIZE);
+    }
+
+    /**
+     * Migrates all consumers of the given anonymous org to the target org.
+     *
+     * @param originOwnerKey key of the source anonymous org
+     * @param destinationOwnerKey key of the destination org
+     */
+    public void migrate(String originOwnerKey, String destinationOwnerKey) {
+        Owner originOwner = this.ownerCurator.getByKey(originOwnerKey);
+        Owner destinationOwner = this.ownerCurator.getByKey(destinationOwnerKey);
+
+        List<String> consumerIds = this.ownerCurator.getConsumerIds(originOwner).list();
```

Comment:
This curator method can be changed to return a Collection directly instead of a CandlepinQuery. The only other instance where it is used(OwnerManager), isn't used for paging.

Reply:
Done. I handled the id gathering a bit differently.

Reply:
I still see the same code here: getConsumerIds returns a CandlepinQuery and then it unpacks it with `.list()`